File: backend/seed-admin.js
// Run once after setting up the database: npm run seed-admin
// Creates an admin login -> username: admin | password: admin123
require('dotenv').config();
const bcrypt = require('bcryptjs');
const pool = require('./config/db');

async function seedAdmin() {
  const username = 'admin';
  const password = 'admin123';

  try {
    const hashed = await bcrypt.hash(password, 10);
    await pool.query('INSERT INTO admins (username, password) VALUES (?, ?)', [username, hashed]);
    console.log('Admin account created.');
    console.log('  username: admin');
    console.log('  password: admin123');
    console.log('Change this password before running a real election.');
  } catch (err) {
    if (err.code === 'ER_DUP_ENTRY') {
      console.log('An admin named "admin" already exists — nothing to do.');
    } else {
      console.error('Could not create admin:', err.message);
    }
  } finally {
    process.exit();
  }
}

seedAdmin();

File: backend/seed_admin.py
# Run once after setting up the database: python seed_admin.py
# Creates an admin login -> username: admin | password: admin123
from app.database import get_conn
from app.auth_utils import hash_password


def seed_admin():
    username = "admin"
    password = "admin123"

    conn = get_conn()
    try:
        with conn.cursor() as cur:
            cur.execute("SELECT id FROM admins WHERE username = %s", (username,))
            if cur.fetchone():
                print('An admin named "admin" already exists — nothing to do.')
                return

            cur.execute(
                "INSERT INTO admins (username, password) VALUES (%s, %s)",
                (username, hash_password(password)),
            )
        conn.commit()
        print("Admin account created.")
        print("  username: admin")
        print("  password: admin123")
        print("Change this password before running a real election.")
    finally:
        conn.close()


if __name__ == "__main__":
    seed_admin()
